pkgs/tools/security/doppler/default.nix

```
@@ -0,0 +1,31 @@
+{ buildGoModule, fetchFromGitHub, stdenv }:
+
+let
+  pname = "doppler";
+  version = "3.9.1";
+  sha256 = "1hfl70k4xz3kxylffybixgklv04sl47knbyd96h8rnmnhscniwzl";
+  vendorSha256 = "0wqbwk72k4r30a3vnf0gnx3k97y8xgnr2iavk5bc8f8vkjv0bsv6";
+in buildGoModule {
+  inherit pname vendorSha256 version;
+
+  src = fetchFromGitHub {
+    inherit sha256;
+
+    owner = "dopplerhq";
+    repo = "cli";
+    rev = version;
+  };
```

`rec` makes this a bit more readable
```suggestion
buildGoModule rec {
  pname = "doppler";
  version = "3.9.1";

  src = fetchFromGitHub {
    owner = "dopplerhq";
    repo = "cli";
    rev = version;
    sha256 = "1hfl70k4xz3kxylffybixgklv04sl47knbyd96h8rnmnhscniwzl";
  };
```

---

This is perhaps mostly a matter of personal preference, but I've come to see `rec`, following Domen Kozar, as an [anti-pattern](https://nix.dev/anti-patterns/language.html#rec-expression). The way I've done it here has the benefit of making it immediately clear which values need to be updated to update the package and cleanly separates declared and computed values.

---

see https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293/4

Essentially `let ... in` is a small `rec` attr set `let a=b; b=a; in ..`, I find them largely interchangeable.

The benefits are marginal, and I find using `rec` improve readability dramatically. It also groups the attrs in their respective callsite, instead of having a flat listing of let bindings.

---

I'll make that change to get this through but I would hope that you (and the community) would reconsider this in the longer term.

---

I'm just one person, and I agree that it's largely preference. The only "hard" fact is that it's "more expensive" to evaluate a `rec` attrset vs a non-rec attrset. However, I'm not sure what the performance is with the let bindings vs a rec attrset

@domenkozar not sure if you have something you would like to contribute to this discusison

---

Okie doke! Definitely an intriguing set of issues. I can also see how in nixpkgs it's probably best to follow standard conventions given maintainability concerns. I'll update this 👍🏼

---

Updated